internal/provider/data_source_http.go

```diff
@@ -97,8 +112,33 @@ func (d *httpDataSource) Read(ctx context.Context, req tfsdk.ReadDataSourceReque
 
 	url := model.URL.Value
 	headers := model.RequestHeaders
+	caCertificate := model.CaCertificate
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
 
-	client := &http.Client{}
+	if !model.Insecure.IsNull() {
+		tr.TLSClientConfig.InsecureSkipVerify = model.Insecure.Value
+	}
+
```

Comment:
Can you use the `SystemCertPool` and if there is an error retrieving it, such as was encountered on Windows, could you log this and create a new cert pool? Perhaps something like the following:

```go
	// Get the System Cert Pool
	caCertPool, err := x509.SystemCertPool()
	if err != nil {
		tflog.Info(ctx, fmt.Sprintf("Error getting copy of system cert pool: %s", err))
		tflog.Debug(ctx, "Creating new cert pool")
		caCertPool = x509.NewCertPool()
	}

	if caCertPool == nil {
		resp.Diagnostics.AddError(
			"Cert pool could not be obtained",
			"Neither the system cert pool or a new cert pool could be obtained",
		)
		return
	}
```

Reply:
I would not use SystemCertPool here for 2 reasons:

1. It results into a different behavior between Windows and Linux. This will result into new issues.
2. If I define an own ca certificate, I would not expect that the system cert pool is append. For example curl has the same behavior too.